packages/doenetml-worker-rust/lib-doenetml-core/src/components/doenet/division.rs

```diff
@@ -417,4 +455,63 @@ mod custom_props {
             }
         }
     }
+
+    pub use division_type::*;
+    mod division_type {
+        use super::*;
+
+        /// The depth of this component in the hierarchy of divisions. E.g., how many
+        /// times this `<section>` is nested in another.
+        #[derive(Debug, Default)]
+        pub struct DivisionTypeProp {}
```

Comment:
If I am reading this correctly, you have a separate `DivisionType` prop that isn't just the `DivisionTypeAttr` prop that is calculated with the general prop update for just this reason: to handle the case where someone omits the `type` attribute and the `<division>` infers its type. Having this explained somewhere would help make it clearer why we have these two props.

Reply:
There is a separate question of whether or not we make this more ergonomic for developers who want to do such tweaking to an attribute. In the JS DoenetML, the equivalent to the `attribute` data query does more processing. It knows the type of the attribute so it actually does something similar to the general prop updaters when you request the attribute, giving you something of the attribute type rather than its components.

I can think of a couple possible avenues we could pursue if we want to go this route, but I guess it's a bit beyond the scope of this PR!

Reply:
Yeah, we can discuss this in another PR :-)